### PR TITLE
Adding service parameters API

### PR DESCRIPTION
**Pull request by albailey**
This also adds unit tests for service param API calls

Signed-off-by: Al Bailey <Al.Bailey@windriver.com>

Prior to starting a PR, please make sure you have read our
[contributor tutorial](https://github.com/gophercloud/gophercloud/tree/master/docs/contributor-tutorial).

Prior to a PR being reviewed, there needs to be a Github issue that the PR
addresses. Replace the brackets and text below with that issue number.

For #[PUT ISSUE NUMBER HERE]

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

[PUT URLS HERE]

### CODE REVIEW

File: starlingx/inventory/v1/serviceparameters/requests.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+		return r
+	}
+
+	_, r.Err = c.Post(createURL(c), reqBody, &r.Body, &gophercloud.RequestOpts{
+		OkCodes: []int{200, 201, 202},
+	})
+	return r
+}
+
+// Apply applies the service parameters through a POST. Optionally a service name can be passed'
```
Comment (donpenney) [inline starlingx/inventory/v1/serviceparameters/requests.go]: nit: Remove the trailing ' character

File: starlingx/inventory/v1/serviceparameters/testing/fixtures.go [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+		ParamName:  "http_port",
+		ParamValue: "8080",
+	}
+
+	bbqUUID             = "75b3757d-4168-43c8-b921-5231de1955c9"
+	bbqResource         = "bbq::brickets::charcoal::mode"
+	ServiceParameterBBQ = serviceparameters.ServiceParameter{
+		ID:         "75b3757d-4168-43c8-b921-5231de1955c9",
+		Service:    "bbq",
+		Section:    "brickets",
```
Comment (donpenney) [inline starlingx/inventory/v1/serviceparameters/testing/fixtures.go]: spelling: briquettes

**Review by donpenney — APPROVED** [review]

**Review by mpeters-wrs — APPROVED** [review]

File: starlingx/inventory/v1/serviceparameters/requests.go (hunk 1 above)
Comment (albailey) [reply starlingx/inventory/v1/serviceparameters/requests.go]: fixing

File: starlingx/inventory/v1/serviceparameters/testing/fixtures.go (hunk 2 above)
Comment (albailey) [reply starlingx/inventory/v1/serviceparameters/testing/fixtures.go]: fixing

**Comment by albailey** [comment]
Let me know if I need to create a new pull request that squashes those commits together, or if this pull request is smart enough to squash them together

**Review by donpenney — APPROVED** [review]
LGTM

**Pull request merged**